storageinsights/src/test/java/com/google/cloud/storage/storageinsights/samples/test/ITStorageinsightsSamplesTest.java

```diff
@@ -0,0 +1,232 @@
+import com.google.cloud.storageinsights.v1.LocationName;
+import com.google.cloud.storageinsights.v1.ReportConfig;
+import com.google.cloud.storageinsights.v1.StorageInsightsClient;
+import com.google.cloud.testing.junit4.StdOutCaptureRule;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class ITStorageinsightsSamplesTest {
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String SINK_BUCKET = "insights-test-bucket-sink" + UUID.randomUUID();
+  private static final String SOURCE_BUCKET = "insights-test-bucket-source" + UUID.randomUUID();
+  public static final String BUCKET_LOCATION = "us-west1";
+  private static Storage storage;
+  private static StorageInsightsClient insights;
+
+  @Rule public final StdOutCaptureRule stdOutCaptureRule = new StdOutCaptureRule();
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    insights = StorageInsightsClient.create();
+
+    storage = StorageOptions.newBuilder().build().getService();
+    storage.create(
+        BucketInfo.newBuilder(SOURCE_BUCKET)
+            .setLocation(BUCKET_LOCATION)
+            .setLifecycleRules(
+                ImmutableList.of(
+                    new BucketInfo.LifecycleRule(
+                        BucketInfo.LifecycleRule.LifecycleAction.newDeleteAction(),
+                        BucketInfo.LifecycleRule.LifecycleCondition.newBuilder()
+                            .setAge(1)
+                            .build())))
+            .build());
+    storage.create(
+        BucketInfo.newBuilder(SINK_BUCKET)
+            .setLocation(BUCKET_LOCATION)
+            .setLifecycleRules(
+                ImmutableList.of(
+                    new BucketInfo.LifecycleRule(
+                        BucketInfo.LifecycleRule.LifecycleAction.newDeleteAction(),
+                        BucketInfo.LifecycleRule.LifecycleCondition.newBuilder()
+                            .setAge(1)
+                            .build())))
+            .setStorageClass(StorageClass.NEARLINE)
+            .build());
+
+    ProjectsClient pc = ProjectsClient.create();
+    Project project = pc.getProject(ProjectName.of(PROJECT_ID));
+    String projectNumber = project.getName().split("/")[1];
+    String insightsServiceAccount =
+        "service-" + projectNumber + "@gcp-sa-storageinsights.iam.gserviceaccount.com";
+
+    grantBucketsInsightsPermissions(insightsServiceAccount, SOURCE_BUCKET);
+    grantBucketsInsightsPermissions(insightsServiceAccount, SINK_BUCKET);
+    Thread.sleep(10000); //gives time for service account permissions to propagate.
```

Comment:
nit: i recommend to follow up on this. adding 10 seconds for each check "just in case" is not a good solution. did you consider using externally provisioned SA?